source_type.go

```diff
@@ -0,0 +1,46 @@
+package grpc_prometheus
+
+import (
+	"context"
+	"strconv"
+)
+
+type sourceContextKeyType struct{}
+
+// sourceContextKey unique key for all packages
+var sourceContextKey sourceContextKeyType
+
+type Source int
+
+const (
+	// Internal for requests that have header referer
+	Internal Source = 0
+
+	// External for other requests
+	External Source = 1
+)
+
+func (s Source) String() string {
+	switch s {
+	case Internal:
+		return "internal"
+	case External:
+		return "external"
+	default:
+		return "source(" + strconv.FormatInt(int64(s), 10) + ")"
```

Comment:
Why do we need it if other sources are not supported?

Reply:
Because we can make some like Source(123)

Same solution in Status::String()

Reply:
My point is that we control this values, and I rather panic. Because it is not expected.
But ok. Let's leave it as it is